recipe-client-addon/test/unit/test_NormandyApi.js

```diff
@@ -126,6 +142,73 @@ add_task(withMockApiServer(async function test_fetchRecipes() {
   equal(recipes[0].name, "system-addon-test");
 }));
 
+add_task(async function test_fetchRecipes_canonical_mismatch() {
+  const getApiUrl = sinon.stub(NormandyApi, "getApiUrl").resolves("http://localhost/recipes/");
+
+  // Recipe is non-canonical (it has whitespace, properties are out of order)
+  const response = new MockResponse(`[
+    {
+      "recipe": {"b": 1, "a": 2},
+      "signature": {"signature": "", "x5u": ""}
+    }
+  ]`);
+  const get = sinon.stub(NormandyApi, "get").resolves(response);
+
+  try {
+    await NormandyApi.fetchRecipes();
+    ok(false, "fetchRecipes did not throw for canonical JSON mismatch");
+  } catch (err) {
+    ok(err instanceof InvalidSignatureError, "Error was not an InvalidSignatureError");
```

Comment:
Nitty nit: personally, I prefer phrasing assertion messages as "X should Y" rather than "X is Y" or "X is not Y", because reading passing test messages is confusing in the "X is not Y" case (and "X is Y" is confusing in the failure case, though less so). That is, if I look at this test passing and it says:

```
TEST-PASS | Error was not an InvalidSignatureError
```
I would assume that the error not being an invalid signature error was the right thing. Especially if there are then failures later in the test, the negations cause me to make wrong assumptions about the state of the program prior to the actual test failures.

Anyway, I recognize this is potentially mostly personal preference, so if you want to leave it like this that wfm.